CONTRIBUTING.md

```diff
@@ -178,6 +179,80 @@ If notebooks fail because of changes to pybamm, it can be a bit of a hassle to d
 python run-tests.py --debook examples/notebooks/notebook-name.ipynb script.py
 ```
 
+### Debugging
+
+Often, the code you write won't pass the tests straight away, at which stage it will become necessary to debug.
+The key to successful debugging is to isolate the problem by finding the smallest possible example that causes the bug.
+In practice, there are a few tricks to help you to do this, which we give below.
+Once you've isolated the issue, it's a good idea to add a unit test that replicates this issue, so that you can easily check whether it's been fixed, and make sure that it's easily picked up if it crops up again.
+This also means that, if you can't fix the bug yourself, it will be much easier to ask for help (by opening a [bug-report issue](https://github.com/pybamm-team/PyBaMM/issues/new?template=bug_report.md)).
+
+1. Run individual test scripts instead of the whole test suite:
+```bash
+python3 tests/unit/path/to/test
+```
+You can further restrict which tests are run from a particular script by using the skipping decorator:
+```python3
+@unittest.skip("")
+def test_bit_of_code(self):
+    ...
+```
+or by just commenting out all the tests you don't want to run
```

Comment:
can also run a single test by providing the name of the test you want to run, eg.

```bash
python tests/unit/test_quick_plot.py TestQuickPlot.test_failure
```